src/Provider/Intercom.php

```diff
@@ -119,6 +125,10 @@ protected function fetchResourceOwnerDetails(AccessToken $token)
      */
     protected function createResourceOwner(array $response, AccessToken $token)
     {
-        return new IntercomResourceOwner($response);
+        $validated_response = $response;
+        if ($this->verify_email == true && $response['email_verified'] != true) {
+            $validated_response = array();
```

Comment:
To make this future-proof, I'd use array square bracket syntax `$validatedResponse = [];`